app/models/user.rb

```diff
@@ -21,3 +22,7 @@ def as_json(options = {})
     super(options.merge(except: [:created_at, :updated_at, :password_digest]))
   end
 end
+
+def set_default_role
+  self.role ||= Role.find_by(name: 'CUSTOMER')
```

Comment:
If you are doing it here, then why in controller again?
Move 'Customer' to constant